spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -385,12 +400,21 @@ def main(argv=None):
                                                             group_funcs=group_funcs)
     metrics_agg_merged = merge_dict(metrics_agg)
     save_as_csv(metrics_agg_merged, file_out, fname_in=fname_segmentation, append=append)
-
-    # QC report (only show CSA for clarity)
+    # QC report (only for PMJ-based CSA)
     if path_qc is not None:
-        generate_qc(fname_segmentation, args=arguments, path_qc=os.path.abspath(path_qc), dataset=qc_dataset,
-                    subject=qc_subject, path_img=_make_figure(metrics_agg_merged, fit_results),
-                    process='sct_process_segmentation')
+        if fname_pmj is not None:
+            if arguments.qc_image is not None:
+                generate_qc(fname_in1=get_absolute_path(arguments.qc_image),
+                            fname_seg=[fname_ctl_smooth, fname_pmj, fname_mask_out, fname_ctl_smooth],  # Repeat centerline to flatten image along the centerline
```

Comment:
Here is a more verbose comment that explains the entire situation!

```suggestion
                            # NB: For this QC figure, the centerline has to be first in the list in order for the PMJ
                            # marker to be properly layered on top of the centerline. However, Sagittal.get_center_spit
                            # is called during QC, and it uses `fname_seg[-1]` to center the slices. `fname_mask_out`
                            # doesn't work for this, so we have to repeat `fname_ctl_smooth` at the end of the list.
                            fname_seg=[fname_ctl_smooth, fname_pmj, fname_mask_out, fname_ctl_smooth], 
```